libbeat/template/load.go

```diff
@@ -0,0 +1,109 @@
+package template
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/paths"
+)
+
+// TemplateLoader is a subset of the Elasticsearch client API capable of
+// loading the template.
+type ESClient interface {
+	LoadJSON(path string, json map[string]interface{}) ([]byte, error)
+	Request(method, path string, pipeline string, params map[string]string, body interface{}) (int, []byte, error)
+	GetVersion() string
+}
+
+type Loader struct {
+	config   TemplateConfig
+	client   ESClient
+	beatInfo common.BeatInfo
+}
+
+func NewLoader(cfg *common.Config, client ESClient, beatInfo common.BeatInfo) (*Loader, error) {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function NewLoader should have comment or be unexported